src/components/MovieList/index.js

```diff
@@ -1,18 +1,41 @@
 import MovieListItem from "../MovieListItem";
+import { useState } from "react";
+import MovieGallery from "../MovieGallery";
 
 const MovieList = ({
                        lists = [{
                            _id: "1",
                            listName: "My List",
                            movies: []
-                       }], listItemOnClickHandler
+                       }]
                    }) => {
+    const [showListDetails, setShowListDetails] = useState(false);
+    const [listDetails, setListDetails] = useState({});
+    const listClickHandler = (list) => {
+        setShowListDetails(true);
+        setListDetails(list);
+    }
+    const goBackClickHandler = () => {
+        setShowListDetails(false);
+        setListDetails({});
+    }
     return (
-        <div>
+        <>
+        
+            { !showListDetails && 
+            <div className="row g-2 mt-4 mb-4">
+                {lists && lists.map(l => <MovieListItem key={l._id} list={l} onClickHandler={listClickHandler}/>)}
+            </div>
+            }
             {
-                lists && lists.map(l => <MovieListItem key={l._id} list={l} onClickHandler={listItemOnClickHandler}/>)
+            showListDetails &&
+            <>
+            <div className="fs-4 mb-2" onClick={goBackClickHandler}><i class="fas fa-angle-left"></i> <span className="align-items-center">Go Back</span></div>
+            <MovieGallery movies={listDetails.movies} posterOnClickHandler={() => {}}/>
```

Comment:
Can you also add the ```posterOnClickHandler``` so that when clicking the movie the user will navigate to the movie details page?
I believe the route is ```details/:mid```.

Reply:
Do you mean when clicking each movie inside the list?

Reply:
Yes, click and we go to the movie details screen

Reply:
But, is that onclick being passed from the parent?

Reply:
Or do I create a new one?

Reply:
I think you can directly add it here, as an inline function.

Reply:
Added that functionality. Can you check again? Thanks a lot!